docs/DesignDocs/ExtendedEnclaveInitializationData.md

```diff
@@ -0,0 +1,154 @@
+Extended Enclave Initialization Data
+=====
+
+We present an extension of Intel SGX attestation in OpenEnclave that enables the attestation of enclave initialization parameters.
+Our extension uses MRENCLAVE as a platform configuration register to measure additional data besides the signed enclave image, such as instance-specific enclave configuration and the settings in the current enclave signing configuration.
+
+We introduce a new OE quote type for SGX quotes with extended enclave initialization data (EEID).
+Recipients of such quotes must be able to recover and verify the original enclave identity information (including the original MRENCLAVE and MRSIGNER of the base image), based on the knowledge of the initialization data, which is transmitted alongside the quote as a new collateral type.
+
+
+Motivations
+----------
+
+With current SGX quotes, applications can attest their configuration as part of the `reportData` argument of `oe_get_report`.
```

Comment:
We need to decouple discussion of evidence format from discussion of APIs.
APIs are supposed to be evidence format agnostic, not evidence format specific.
If you create a new evidence format, that's ok, but I think we need to find a way to have a generic API that is format agnostic and not add a new API per format.
(Note: oe_get_report is named poorly since it uses an SGX-specific name, rather than the IETF-agnostic name, and should really be oe_get_evidence.  Maybe we will rename it to that in the future?)

Reply:
Absolutely, the APIs should be as generic as possible. We can put the EEID into `oe_enclave_setting_t` so that it doesn't require a change to the API function itself. I think the verification can go into a verifier plugin, so that the API surface doesn't change.